Source/Utils/WarpXUtil.cpp

```diff
@@ -478,6 +478,23 @@ getArrWithParser (const amrex::ParmParse& a_pp, char const * const str, std::vec
     }
 }
 
+#if defined(BL_USE_FLOAT) != defined(AMREX_SINGLE_PRECISION_PARTICLES)
+void
+getArrWithParser (const amrex::ParmParse& a_pp, char const * const str, std::vector<amrex::ParticleReal>& val,
+                    const int start_ix, const int num_val)
```

Comment:
Granted I'm really bad at modern C++, but is there a way to use a template function instead so that this automatically works with both `Real` and `ParticleReal` versions rather than having to repeat the full code?
Also in this case `val` is `static_cast` to `Real` which seems fishy...

Reply:
I redefined the function as it is called multiple times with val being a vector of `Real` or `ParticleReal` values, so having only one definition errors out if `Real` and `ParticleReal` are different. As for why it is `static_cast`to `Real`, I'm guessing it's because the next operations done on it expect it to be `Real`.